app/service/url_shortener.py

```diff
@@ -0,0 +1,28 @@
+from fastapi import APIRouter, HTTPException
+from pydantic import BaseModel
+from app.models.database import create_link, get_link_by_short_url
+
+#this file handles the logic of interacting and modifying the database
+router = APIRouter()
+
+#ShortenRequest ensures the input is in the correct format (JSON)
```

Comment:
How does this class verify that is in JSON?

Reply:
In this version, ShortenRequest uses BaseModel, which is a pydantic feature to define what the object should look like, and validates requests to be the same format.
This function has been renamed and moved to database.py for now